### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by mxmlnkn**
not sure why the pull requests from master appear in the commit list ... But it seems like it doesn't matter, because it is merged instead of rebased, so only the final state of this bugfix-branch counts. Maybe we could set up github to rebase instead of merge. It's confusing, or I rebase onto dev, because it seems I branched from master, that's why this happened.

### CODE REVIEW

File: src/imresh/io/writeOutFuncs/writeOutFuncs.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ namespace writeOutFuncs
                     if ( isnan(value) or isinf(value) or value < 0 )
                     {
                         /* write out red  pixel to alert user that something is wrong */
-                        png.plot( 1+ix, 1+iy, 65535, 0, 0 );
+                        png.plot( 1+ix, 1+iy, 65535*isnan(value), 65535*isinf(value), 65536*(value < 0) );
```
Comment (Ferruck) [inline src/imresh/io/writeOutFuncs/writeOutFuncs.cpp:98]: Mind the style guide! :smile:

File: src/imresh/io/writeOutFuncs/writeOutFuncs.cpp [hunk 2]
```diff
@@ -111,16 +111,23 @@ namespace writeOutFuncs
...
+        void writeOutAndFreePNG
+        (
+            float * _mem,
+            std::pair<unsigned int,unsigned int> const _size,
+            std::string const _filename
+        )
+        {
+            writeOutPNG( _mem, _size, _filename );
+            justFree( _mem, _size, _filename );
+        }
```
Comment (Ferruck) [inline src/imresh/io/writeOutFuncs/writeOutFuncs.cpp:130]: Nice idea, providing both a freeing and a non-freeing version.

File: tests/imresh/io/testReadWrite.cpp [hunk 3]
```diff
@@ -102,8 +102,8 @@ namespace io
                 for ( auto i = 0; i < nElements; ++i )
                     pData[i] = (float) rand() / RAND_MAX;
 
-                /* note that this call deletes the pointer @todo: move delete[] inside taskQueue.cu */
-                writeOutPNG( pData, ImageDim{ Nx, Ny }, tmpFileName );
+                writeOutAndFreePNG( pData, ImageDim{ Nx, Ny }, tmpFileName );
+                delete[] pData;
```
Comment (Ferruck) [inline tests/imresh/io/testReadWrite.cpp:106]: But now I don't get why you'd have to call `delete[]` after you've called a `writeOutAndFree*` function.

**Comment by Ferruck** [comment]
You don't have to set up Github to rebase instead of merging. That would be a question of how you're using git on the command line.
The main problem here is that you've branched from `master` which you shouldn't do.
The workflow is:
1. Branch `dev`
2. Make changes
3. Open up a pull request to merge back your changes into `dev`
4. When `dev` is in a state where it's kind of production ready and has enough changes since the last release, merge `dev` into `master` and make a new release (by tagging the `master` branch)
5. Start again with 1.

You could work with rebasing but that leads to an ugly repository history. [Here](http://nvie.com/posts/a-successful-git-branching-model/) is a nice read on how we (or at least I) try to work with git.

File: tests/imresh/io/testReadWrite.cpp (hunk 3 above)
Comment (mxmlnkn) [inline tests/imresh/io/testReadWrite.cpp:106]: This is wrong here :S, thanks

**Pull request merged**